go/pools/resource_pool.go

```diff
@@ -104,9 +104,12 @@ func (rp *ResourcePool) TryGet() (resource Resource, err error) {
 
 func (rp *ResourcePool) get(ctx context.Context, wait bool) (resource Resource, err error) {
 	// If ctx has already expired, avoid racing with rp's resource channel.
-	if ctx.Err() != nil {
+	select {
+	case <-ctx.Done():
```

Comment:
The open-source Context docs don't seem to mention it, but the return value of Err() is unspecified if you call it before the context's Done() channel is closed.